x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -161,431 +161,574 @@ export const ApiKeyFlyout: FunctionComponent<ApiKeyFlyoutProps> = ({
-    </FormFlyout>
+            <EuiSpacer />
+            <EuiFormFieldset>
+              <EuiSwitch
+                label={
+                  <FormattedMessage
+                    id="xpack.security.accountManagement.apiKeyFlyout.includeMetadataLabel"
+                    defaultMessage="Include metadata"
+                  />
+                }
+                data-test-subj="apiKeysMetadataSwitch"
+                checked={formik.values.includeMetadata}
+                disabled={readOnly || (apiKey && !canEdit)}
+                onChange={(e) => formik.setFieldValue('includeMetadata', e.target.checked)}
+              />
+              {formik.values.includeMetadata && (
+                <>
+                  <EuiSpacer size="m" />
+                  <FormRow
+                    data-test-subj="apiKeysMetadataCodeEditor"
+                    helpText={
+                      <DocLink
+                        app="elasticsearch"
+                        doc="security-api-create-api-key.html#security-api-create-api-key-request-body"
+                      >
+                        <FormattedMessage
+                          id="xpack.security.accountManagement.apiKeyFlyout.metadataHelpText"
+                          defaultMessage="Learn how to structure metadata."
+                        />
+                      </DocLink>
+                    }
+                    fullWidth
+                  >
+                    <FormField
+                      as={CodeEditorField}
+                      name="metadata"
+                      options={{ readOnly: readOnly || (apiKey && !canEdit) }}
+                      value={formik.values.metadata}
+                      onChange={(value: string) => formik.setFieldValue('metadata', value)}
+                      validate={(value: string) => {
+                        if (!value) {
+                          return i18n.translate(
+                            'xpack.security.management.apiKeys.apiKeyFlyout.metadataRequired',
+                            {
+                              defaultMessage: 'Enter metadata or disable this option.',
+                            }
+                          );
+                        }
+                        try {
+                          JSON.parse(value);
+                        } catch (e) {
+                          return i18n.translate(
+                            'xpack.security.management.apiKeys.apiKeyFlyout.invalidJsonError',
+                            {
+                              defaultMessage: 'Enter valid JSON.',
+                            }
+                          );
+                        }
+                      }}
+                      fullWidth
+                      languageId="xjson"
+                      height={200}
+                    />
+                  </FormRow>
+                  <EuiSpacer size="s" />
+                </>
+              )}
+            </EuiFormFieldset>
+          </Form>
```

Comment:
I noticed that if you quickly hit return twice in the flyout it will create two API keys with the same configuration. I checked against main and it behaves the same way, so this was an existing issue.